pkg/dataplane/client.go

```diff
@@ -2,186 +2,96 @@ package dataplane
-	}
+var _ Client = (*clientAdapter)(nil)
 
-	creds, err := c.swaggerClient.Getcreds(ctx, credRequestDef, nil)
+func (c *clientAdapter) DeleteSystemAssignedIdentity(ctx context.Context) error {
+	resp, err := c.delegate.DeleteidentityWithResponse(ctx, &internal.DeleteidentityParams{ApiVersion: internal.DeleteidentityParamsApiVersion(apiVersion)})
 	if err != nil {
-		return nil, fmt.Errorf("%w: %w", errGetCreds, err)
+		return err
 	}
-
-	if err := validateUserAssignedMSIs(creds.ExplicitIdentities, request.ResourceIDs); err != nil {
-		return nil, err
-	}
-
-	// Tenant ID is a header passed to RP frontend, so set it here if it's not set
-	for _, identity := range creds.ExplicitIdentities {
-		if *identity.TenantID == "" {
-			*identity.TenantID = request.TenantID
+	for _, respErr := range []*internal.ErrorResponse{resp.JSON400, resp.JSON401, resp.JSON403, resp.JSON404, resp.JSON405, resp.JSON429, resp.JSON500, resp.JSON503} {
+		if respErr != nil {
+			return &ResponseError{WrappedError: *respErr}
 		}
 	}
-
-	credentialsObject := CredentialsObject{Values: creds.CredentialsObject, cloud: c.cloud}
-	if !credentialsObject.IsUserAssigned() {
-		return nil, errNoUserAssignedMSIs
-	}
-	return &credentialsObject, nil
+	return nil
 }
 
-func (c *ManagedIdentityClient) GetCredentialsObjectUserAssignedIdentities(ctx context.Context, request UserAssignedMSIRequest) (*CredentialsObject, error) {
-	return c.getUserAssignedIdentities(ctx, request)
-}
-
-func (c *ManagedIdentityClient) GetNestedCredentialsObjectUserAssignedIdentities(ctx context.Context, request UserAssignedMSIRequest) (*NestedCredentialsObject, error) {
-	ua, err := c.getUserAssignedIdentities(ctx, request)
+func (c *clientAdapter) GetSystemAssignedIdentityCredentials(ctx context.Context) (*ManagedIdentityCredentials, error) {
+	resp, err := c.delegate.GetcredWithResponse(ctx, &internal.GetcredParams{ApiVersion: internal.GetcredParamsApiVersion(apiVersion)})
 	if err != nil {
 		return nil, err
 	}
-
-	if len(ua.Values.ExplicitIdentities) == 0 ||
-		ua.Values.ExplicitIdentities[0] == nil {
-		return nil, errGetNestedCreds
+	for _, respErr := range []*internal.ErrorResponse{resp.JSON400, resp.JSON401, resp.JSON403, resp.JSON404, resp.JSON429, resp.JSON500, resp.JSON503} {
+		if respErr != nil {
+			return nil, &ResponseError{WrappedError: *respErr}
+		}
 	}
-
-	return &NestedCredentialsObject{Values: *ua.Values.ExplicitIdentities[0], cloud: c.cloud}, nil
+	return resp.JSON200, nil
 }
 
-func validateResourceIDs(fl validator.FieldLevel) bool {
-	field := fl.Field()
-
-	// Confirm we have a slice of strings
-	if field.Kind() != reflect.Slice {
-		return false
-	}
-
-	if field.Type().Elem().Kind() != reflect.String {
-		return false
-	}
-
-	// Check we have at least one element
-	if field.Len() < 1 {
-		return false
+func (c *clientAdapter) GetUserAssignedIdentitiesCredentials(ctx context.Context, request UserAssignedIdentitiesRequest) (*ManagedIdentityCredentials, error) {
```

Comment:
The previous behaviour used to run the validations  https://github.com/Azure/msi-dataplane/blob/main/pkg/dataplane/client.go#L98 (e.g all the requested resourcesIds are there). Will that behaviour still be maintained?

Reply:
We have no reason to be "validating" what the authoritative server gave us as a response. That logic is removed on purpose.

Reply:
The prev version of the client guaranteed this if at least one of the request resourceId doesn't exist, a user of the client will get an error. Do we know what response does the authoritative server return in such cases?